File: 9. 回文数/swift/main/main/main.swift
//
//  main.swift
//  main
//
//  Created by cedricwu on 7/16/25.
//

import Foundation

class Solution {
    func isPalindrome(_ x: Int) -> Bool {
        let list = Array(String(x))
        var left = 0
        var right = list.count - 1
        while left < right {
            if list[left] == list[right] {
                left += 1
                right -= 1
            } else {
                return false
            }
        }
        return true
    }
}

let s = Solution()
print(s.isPalindrome(-121))

File: 9. 回文数/rust/main/src/main.rs
struct Solution {}

impl Solution {
    pub fn is_palindrome(x: i32) -> bool {
        let list: Vec<char> = x.to_string().chars().collect();
        let mut left = 0;
        let mut right = list.len() - 1;
        while left < right {
            if list[left] == list[right] {
                left += 1;
                right -= 1;
            } else {
                return false
            }
        }
        true
    }
}

fn main() {
    println!("Hello, world!");
}
